cfme/tests/candu/test_host_graph.py

```diff
@@ -0,0 +1,90 @@
+import pytest
+
+from cfme import test_requirements
+from cfme.infrastructure.provider.rhevm import RHEVMProvider
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.wait import wait_for
+
+
+pytestmark = [
+    pytest.mark.tier(3),
+    test_requirements.c_and_u,
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([VMwareProvider, RHEVMProvider],
+                         required_fields=[(['cap_and_util', 'capandu_vm'], 'cu-24x7')])
+]
+
+
+HOST_RECENT_HR_GRAPHS = ['host_cpu',
+                         'host_memory',
+                         'host_disk',
+                         'host_network']
+
+
+@pytest.fixture(scope='function')
+def host(appliance, provider):
+    collection = appliance.collections.hosts
+    for test_host in provider.data['hosts']:
+        if not test_host.get('test_fleece', False):
+            continue
+        return collection.instantiate(name=test_host.name, provider=provider)
+
+
+@pytest.mark.uncollectif(lambda provider, graph_type:
+                         provider.one_of(RHEVMProvider) and
+                         graph_type == "host_disk")
+@pytest.mark.parametrize('graph_type', HOST_RECENT_HR_GRAPHS)
+def test_host_most_recent_hour_graph_screen(graph_type, provider, host, enable_candu):
+    """ Test Host graphs for most recent hour displayed or not
+
+    prerequisites:
+        * C&U enabled appliance
+
+    Steps:
+        * Navigate to Host Utilization Page
+        * Check graph displayed or not
+        * Check legends hide and display properly or not
+        * Check data for legends collected or not
+    """
+
+    host.wait_candu_data_available(timeout=1200)
+
+    view = navigate_to(host, 'candu')
+    view.options.interval.fill('Most Recent Hour')
+
+    if graph_type == 'host_cpu':
+        graph = view.interval_type.host_cpu
+    elif graph_type == 'host_memory':
+        graph = view.interval_type.host_memory
+    elif graph_type == 'host_disk':
+        graph = view.interval_type.host_disk
+    else:
```

Comment:
**OPTIONAL**: Personally, I'd like to see either a ```host_network``` inside an ```elif``` and an ```else``` declaring that it's not a valid type, just in case. In all honesty, I'd love to see this being a dictionary. I know it's difficult because we need to have the ```view``` before we can get them. We could use a chain of ```getattr``` with the dict value. However I am happy to merge this without. Please let me know.

Reply:
I like to try suggestions. Please wait.

Reply:
@psav: I removed total conditions. just used 
` graph = getattr(view.interval_type, graph_type)`

I found no need of passing `dict`. As we are passing same `str` which are attributes of `view.interval`. Let me know, your comment on it.